doc/snmp/snmp-configdb-migration-hld.md

```diff
@@ -0,0 +1,221 @@
+# SNMP Migration from snmp.yml to ConfigDB 
+# High Level Design Document
+### Rev 0.1
+
+###### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Travis Van Duyn    | Initial version                   |
+
+# About this Manual
+This document provides general information about the migration of SNMP community information from snmp.yml file to the ConfigDB.
+# Scope
+Currently we are using the /etc/sonic/snmp.yml file to populate the SNMP communities in the /etc/snmp/snmpd.conf file.
+The goal of this update is to move away from the snmp.yml file and move towards the ConfigDB for better supportibility and ease of use.
+
+
+# Config DB
+## SNMP SCHEMA
+Some new "SNMP" tables should be added to ConfigDB in order to store SNMP related configuration. 
+https://github.com/Azure/SONiC/blob/master/doc/snmp/snmp-schema-addition.md
+
+The new SNMP tables are: 
+SNMP
+SNMP_COMMUNITY
+SNMP_USER
+
+
+# SNMP Table
+```
+SNMP|LOCATION
+SNMP|CONTACT
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
+1) "LOCATION"
+2) "Redmond"
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|CONTACT"
+1) "joe"
+2) "joe@microsoft.com"
+```
+
+# SNMP_COMMUNITY Table
+```
+SNMP_COMMUNITY|<community>
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Jack"
+1) "TYPE"
+2) "RW"
+
+```
+
+# SNMP_USER Table 
+```
+SNMP_USER|<user>
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_USER|Travis"
+ 1) "SNMP_USER_ENCRYPTION_TYPE"
+ 2) "AES"
+ 3) "SNMP_USER_AUTH_TYPE"
+ 4) "SHA"
+ 5) "SNMP_USER_ENCRYPTION_PASSWORD"
+ 6) "TravisEncryptPass"
+ 7) "SNMP_USER_AUTH_PASSWORD"
+ 8) "TravisAuthPass"
+ 9) "SNMP_USER_TYPE"
+10) "Priv"
+11) "SNMP_USER_PERMISSION"
+12) "RO"
+```
+
+
+
+# New SNMP CLI Commands
+# Show commands
+```
+admin@str-s6000-acs-11:~$ show run snmp -h
+Usage: show run snmp [OPTIONS] COMMAND [ARGS]...
+
+  Show SNMP information
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community  show runningconfiguration snmp community
```

Comment:
space between "runningconfiguration"

Reply:
In the SONiC CLI, `runningconfiguration` (no spaces) is a subcommand of `show`

Reply:
As this is a help string, looking at the help strings of the existing commands, I thought it would be good to keep it uniform.
 show running --help
Usage: show running [OPTIONS] COMMAND [ARGS]...
  Show current running configuration information
Commands:
  acl         Show acl running configuration

Reply:
Ah, I understand your point, now. The help string is currently just the command syntax. It should be a more human-readable description of what the command does.

Reply:
I updated this document.  This information will be more flushed out during the PR's to submit the commands once we've finalized the details on the design.